localstack/services/cloudformation/models/iam.py

```diff
@@ -164,6 +166,46 @@ def _pre_delete(resource_id, resources, resource_type, func, stack_name):
         }
 
 
+class IAMAccessKey(GenericBaseModel):
+    @staticmethod
+    def cloudformation_type():
+        return "AWS::IAM::AccessKey"
+
+    def get_physical_resource_id(self, attribute=None, **kwargs):
+        return self.props.get("UserName")
+
+    def fetch_state(self, stack_name, resources):
+        user_name = self.resolve_refs_recursively(stack_name, self.props.get("UserName"), resources)
+        keys = aws_stack.connect_to_service("iam").list_access_keys(UserName=user_name)[
+            "AccessKeyMetadata"
+        ]
+
+        return keys[-1]
```

Comment:
hi @dominikschubert could you tell me what you think about this? I'm thinking that the resource Id should be based on the AccessKeyId that is generated after the create_acccess_key operation but I'm not sure how to obtain it or if it's correct.

Reply:
Note: I'm not sure the keys are necessarily have a guaranteed order here so `keys[-1]` as you suspected will not always resolve the correct key. Even with a predictable order this might be wrong (e.g. with multiple access keys for a single user in the same stack when the operations might be interleaved).


What we should do here is set the physical Id manually based on the create response. I'm actually going to quickly add a utility for this, so please be patient for a day, I'll extend this review with the suggestion on how to use this. Shouldn't take longer than a day or two :+1: